### PR TITLE
Add log sink console

### PR DESCRIPTION
**Pull request by dcristoloveanu**

### CODE REVIEW

File: v2/src/log_sink_console.c [hunk 1]
```diff
@@ -1,18 +1,252 @@
...
+                /* Codes_SRS_LOG_SINK_CONSOLE_01_022: [ If any encoding error occurs during formatting of the line (i.e. if any printf class functions fails), log_sink_console.log_sink_log shall print Error formatting log line and return. ]*/
+                result = -1;
+                break;
+            }
+            else
+            {
+                snprintf_result = MIN(snprintf_result, (int)buffer_size);
+                buffer += snprintf_result;
+                buffer_size -= snprintf_result;
+                result += snprintf_result;
```
Comment (ACKPaul) [inline v2/src/log_sink_console.c:70]: Wouldn't it be nice to move these three lines of code to a function that returns  snprintf_result/buffer_size and use it everywhere?

File: v2/src/log_sink_console.c [hunk 2]
```diff
@@ -1,18 +1,252 @@
...
+                    va_list args;
+                    va_start(args, message_format);
+                    int vsnprintf_result = vsnprintf(buffer, buffer_size, message_format, args);
+                    if (vsnprintf_result < 0)
+                    {
+                        /* Codes_SRS_LOG_SINK_CONSOLE_01_022: [ If any encoding error occurs during formatting of the line (i.e. if any printf class functions fails), log_sink_console.log_sink_log shall print Error formatting log line and return. ]*/
+                        error = true;
+                    }
+                    else
+                    {
```
Comment (ACKPaul) [inline v2/src/log_sink_console.c:232]: We can remove the else part .

**Review by ACKPaul — APPROVED** [review]

File: v2/src/log_sink_console.c (hunk 1 above)
Comment (dcristoloveanu) [reply v2/src/log_sink_console.c:70]: Extracted in a macro.

File: v2/src/log_sink_console.c (hunk 2 above)
Comment (dcristoloveanu) [reply v2/src/log_sink_console.c:232]: Added a comment to make it clear that we don't expect to do anything on that branch, but keeping the branch anyway to show it's not missed.

**Pull request merged**